spot-client/src/common/logger/post-to-endpoint.js

```diff
@@ -138,29 +137,28 @@ export default class PostToEndpoint {
             ).then(response => {
                 const { ok, status } = response;
 
-                if (!ok && status >= 500 && status < 600) {
-                    // Retry on server errors and network failures which
-                    // is a regular fetch reject case.
-                    throw new Error(`Request error - status: ${status}`);
-                } else if (ok) {
-                    // Try sending the next request from the queue
-                    this._sendLogs();
-                } else {
+                if (!ok) {
+                    // Throwing will cause a retry on server errors
+                    if (status >= 500 && status < 600) {
+                        throw new Error(`Request error - status: ${status}`);
+                    }
                     // eslint-disable-next-line no-console
                     console.warn(`Dropping log request, status: ${status}`);
                 }
+
+                // Remove the request from the queue and try the next one
+                this._requestsQueue.shift();
+                this._sendLogs();
             })
             .catch(error => {
                 // eslint-disable-next-line no-console
-                console.error(`Log request error, attempt: ${attempt}`, error);
+                console.error(`Log request error, attempt: ${retry + 1}`, error);
 
-                if (attempt < PostToEndpoint.MAX_ATTEMPTS) {
-                    // Put the current request back at the front of the queue
-                    this._requestsQueue.unshift(
-                        new PostLogsRequest(attempt + 1, events));
+                if (retry < PostToEndpoint.MAX_RETRIES) {
```

Comment:
forgot about one thing: now the request must be removed also when the retry limit is exceeded